examples/webgl_mesh_batch.html

```diff
@@ -238,6 +248,25 @@
 			gui.add( api, 'count', 1, MAX_GEOMETRY_COUNT ).step( 1 ).onChange( initMesh );
 			gui.add( api, 'dynamic', 0, MAX_GEOMETRY_COUNT ).step( 1 );
 			gui.add( api, 'method', Method ).onChange( initMesh );
+			gui.add( api, 'opacity', 0, 1 ).onChange( v => {
+
+				if ( v < 1 ) {
+
+					material.transparent = true;
+					material.depthWrite = false;
+
+				} else {
+
+					material.transparent = false;
+					material.depthWrite = true;
+
+				}
+
+				material.opacity = v;
+				material.needsUpdate = true;
```

Comment:
It seems to be the case that "needsUpdate" needs to be called when "transparent" is toggled. Is this intended? This happens on normal Meshes in previous versions of three.js, as well.

Reply:
> It seems to be the case that "needsUpdate" needs to be called when "transparent" is toggled. Is this intended?

Yes, because the `transparent` property controls the `OPAQUE` GLSL define. Without setting  `needsUpdate` to `true` when `transparent` changes, the shaders would not reflect the current material state.

Reply:
That's an implementation detail the user doesn't necessarily have to know about, though - there are plenty of other material fields that don't require setting needsUpdate but trigger a rebuild, including `alphaHash`, `alphaTest`, etc:

https://github.com/mrdoob/three.js/blob/dev/src/renderers/webgl/WebGLPrograms.js#L506-L507